src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -715,6 +715,79 @@ set.subset_set_smul_iff
 
 end pointwise_actions
 
+section
+
+variables {L : Type*} [field L] [algebra K L] (K)
+
+open_locale pointwise
+
+/-- The decomposition subgroup defined as the stabilizer of the action
+on the type of all valuation subrings of the field. -/
+def decomposition_subgroup (A : valuation_subring L) :
+  subgroup (L ≃ₐ[K] L) :=
+mul_action.stabilizer (L ≃ₐ[K] L) A
+
+/-- The valuation subring `A` (considered as a subset of `L`)
+is stable under the action of the decomposition group. -/
+def sub_mul_action (A : valuation_subring L) :
+  sub_mul_action (A.decomposition_subgroup K) L :=
+{ carrier := A,
+  smul_mem' := λ g l h,
+  begin
+    convert set.smul_mem_smul_set h using 1,
+    exact congr_arg coe g.prop.symm,
+end }
+
+/-- The multiplicative action of the decomposition subgroup on `A`. -/
+instance mul_action (A : valuation_subring L) :
+  mul_semiring_action (A.decomposition_subgroup K) A :=
+{ smul_add :=  λ g k l, subtype.ext $ smul_add g k l,
+  smul_zero := λ g, subtype.ext $ smul_zero g,
+  smul_one := λ g, subtype.ext $ smul_one g,
+  smul_mul := λ g k l, subtype.ext $ smul_mul' g k l,
+   ..(sub_mul_action.mul_action (A.sub_mul_action K)) }
```

Comment:
Doesn't
```suggestion
by apply_instance
```
Find this?

Reply:
No, it doesn't seem to work.

Reply:
Hmm. This seems true in more general settings, such as
```lean
instance submonoid.stabilizer_mul_distrib_mul_action
  {α : Type*} [group α] [mul_distrib_mul_action α L] (A : submonoid L) :
  mul_distrib_mul_action (mul_action.stabilizer α A) A :=
{ smul := λ g x, ⟨g • x, (g.prop.le : _ ≤ _) (set.smul_mem_smul_set x.prop)⟩,
  smul_one := λ g, subtype.ext $ smul_one g,
  smul_mul := λ g k l, subtype.ext $ smul_mul' g k l,
  one_smul := λ g, subtype.ext $ one_smul _ g,
  mul_smul := λ g k l, subtype.ext $ mul_smul g k l }
```
but I'm struggling to find a way to reuse this result rather than repeating it for everything extending `submonoid`...